### ISSUE TITLE
Ketrew should be able to execute OCaml workflow nodes

### ISSUE THREAD

**Issue by ihodes** [hammerlab/ketrew#436]
Would be nice if we would serialize OCaml functions and run them as the build process of workflow nodes. I'd much rather be writing OCaml than writing bash or having to add a tool to a machine (dealing with installation etc.) which is not a low-friction activity.

**Comment by hammer** [comment]
Wish GitHub allowed the thinking face emoji for a reaction; this feature would blur the line between workflow managers and distributed execution engines. Definitely an interesting direction to explore.

**Comment by ihodes** [comment]
Yeah, exactly. This is my Trojan horse. It would massively simplify things; we already support this model of execution via our Program module (running bash scripts on workflow nodes).
